### PR TITLE
Fixing 2 issues in GLTFSceneImporter for skinned animations

### PR DESCRIPTION
**Pull request by JohnCopicMS**
And a whole bunch of auto-formatting. Sorry for the noise!

### CODE REVIEW

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs [hunk 1]
```diff
@@ -915,18 +918,21 @@ private BoneWeight[] CreateBoneWeightArray(Vector4[] joints, Vector4[] weights,
...
 		/// </summary>
 		/// <param name="weights">Bone weight array</param>
-		private void NormalizeBoneWeightArray(Vector4[] weights)
+		private void NormalizeBoneWeightArray(Vector4[] weights)
 		{
 			for (int i = 0; i < weights.Length; i++)
 			{
 				var weightSum = (weights[i].x + weights[i].y + weights[i].z + weights[i].w);
-				if (weightSum < 0.98f)
+
```
Comment (JohnCopicMS) [inline UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: This is the first fix: I have animations where, if weighted to 4 bones, has weight values > 1.

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs [hunk 2]
```diff
@@ -949,31 +955,31 @@ protected virtual IEnumerator ConstructMesh(GLTF.Schema.Mesh mesh, Transform par
...
-
-				MeshFilter meshFilter = primitiveObj.AddComponent<MeshFilter>();
+				UnityEngine.Material material = materialCacheData.GetContents(primitive.Attributes.ContainsKey(SemanticProperties.Color(0)));
+
+				UnityEngine.Mesh curMesh = _assetCache.MeshCache[meshId][i].LoadedMesh;
+				if (NeedsSkinnedMeshRenderer(primitive, skin))
+				{
+					var skinnedMeshRenderer = primitiveObj.AddComponent<SkinnedMeshRenderer>();
+					skinnedMeshRenderer.material = material;
+					skinnedMeshRenderer.quality = SkinQuality.Auto;
```
Comment (JohnCopicMS) [inline UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: Second fix: The SkinnedMeshRenderer should not be forced to 4 bones and ignore the project's quality settings.

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs [hunk 3]
```diff
@@ -861,40 +864,40 @@ private bool HasBones(Skin skin)
...
-
-			GLTFHelpers.BuildBindPoseSamplers(ref attributeAccessor);
-
-			GLTF.Math.Matrix4x4[] gltfBindPoses = attributeAccessor.AccessorContent.AsMatrix4x4s;
-			Matrix4x4[] bindPoses = new Matrix4x4[skin.Joints.Count];
-
+			Transform[] bones = new Transform[boneCount];
+
+			int bufferId = skin.InverseBindMatrices.Value.BufferView.Value.Buffer.Id;
+			//BufferCacheData bufferCacheData = _assetCache.BufferCache[bufferId];
```
Comment (blgrossMS) [inline UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: ?

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs (hunk 1 above)
Comment (blgrossMS) [reply UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: Should you use an epsilon value?

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs (hunk 2 above)
Comment (blgrossMS) [reply UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: Yeah I left that note on the previous PR. I think it should be a setting.

Comment (JohnCopicMS) [reply UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: For now I think leaving it on Auto (like the fbx importer) is fine and a setting could be added in the future.

Comment (blgrossMS) [reply UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: Works for me

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs (hunk 1 above)
Comment (JohnCopicMS) [reply UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: Yes!

**Pull request merged**